ledger/eval/eval.go

```diff
@@ -948,13 +952,16 @@ func (eval *BlockEvaluator) TransactionGroup(txgroup []transactions.SignedTxnWit
 	defer cow.recycle()
 
 	evalParams := logic.NewEvalParams(txgroup, &eval.proto, &eval.specials)
+	evalParams.GranularEval = eval.GranularEval
 	evalParams.Tracer = eval.Tracer
 
 	if eval.Tracer != nil {
 		eval.Tracer.BeforeTxnGroup(evalParams)
 		// Ensure we update the tracer before exiting
 		defer func() {
-			eval.Tracer.AfterTxnGroup(evalParams, err)
+			// TODO: need to check, is it ok to call .updates() after .commitToParent()?
```

Comment:
I agree it's important to make sure that the ordering of these is interchangeable. I'm also worried about proving that the final delta produced from the block as a whole (which is passed to the trackers) is identical when you're splitting out these COWs. Some tests there would help me be more confident in that.

Reply:
Added unit test to ensure it's ok to call `deltas` after `commitToParent`. Also added tests to ensure the deltas being received by tracers are accurate. Note, these won't be the exact same as the StateDelta at the end of a block, since the tracers receive these deltas before additional end of block processing happens